packages/block-library/src/site-logo/edit.js

```diff
@@ -530,9 +582,44 @@ export default function LogoEdit( {
 
 	const label = __( 'Add a site logo' );
 
+	const mediaInspectorPanel = ( canUserEdit || logoUrl ) && (
+		<InspectorControls>
+			<PanelBody title={ __( 'Media' ) }>
+				<div className="block-library-site-logo__inspector-media-replace-container">
+					{ ! canUserEdit && !! logoUrl && <InspectorLogoPreview /> }
+					{ canUserEdit && !! logoUrl && (
+						<SiteLogoReplaceFlow
+							{ ...mediaReplaceFlowProps }
+							name={ <InspectorLogoPreview /> }
```

Comment:
It's slightly surprising that this just works; I guess it's more of a "label element" than a name.

The destination component, `MediaReplaceFlow`, doesn't type its props in the code but it does document them in its README file. It turns out that [the name is supposed to be a string](https://github.com/WordPress/gutenberg/blob/5c1d6ba207fce7c06ad94d51a62c204ae34d82f7/packages/block-editor/src/components/media-replace-flow/README.md#name). What should we do here? Leave that component untouched and work around? Or amend `MediaReplaceFlow` to explicitly accept elements? I wonder if it's safer to add a separate prop (like `label`) or just make `name` polymorphic. I have no opinion here; maybe the latter option is enough.

Reply:
Oh yes great catch! I forgot to add the comment here yesterday and cc @ciampo and @mirka about this. Right now is as you say a workaround, without adding new props. We could deprecate 'name' and add a new polymorphic prop to handle this with a better name(no pun intended).

Reply:
It's working because internally `name` is just interpolated in the JSX:

https://github.com/WordPress/gutenberg/blob/c6029852e1bd5af06aa87d55a6789620f7c6a727/packages/block-editor/src/components/media-replace-flow/index.js#L147-L155

Under a technical point of view, it feels safe to change the docs and allow `name` to be any `ReactNode`. I guess it depends more on whether you'd prefer to "limit" consumers of the component to pass strings of text for design / semantics / any other reasons.